medienhaus/medienhaus-spaces
# Display login prompt when service token is invalid

### Changes made

- Added 'use-immer' for improved state management.
- Introduced a 'connectionStatus' object to manage service connection statuses.
- Implemented 'makeRequest' methods within auth providers to handle specific response statuses (401 and 403).
- Modified the 'LoginPrompt' component to display an error message and password `form` when the service connection status is 'false'.

### Testing

- login to spaces
- invalidate your `etherpad` and/or `spacedeck` tokens by for example manipulating them directly in the account data object 
- navigate to /write or /spacdeck
- login prompt should be displayed
- enter password and confirm

![Screen Shot 2023-11-22 at 14 45 51](https://github.com/medienhaus/medienhaus-spaces/assets/706419/e7824e0a-f79f-44e1-aa9e-5c8239b2612e)
